### PR TITLE
Mileage

### LINKED ISSUE #86: The user should be able to add Mileage
[Figma](https://www.figma.com/file/7HZJK17nvKpicBO3SuRWt7/Expense-%26-Pay2?node-id=1049%3A64196)

**Requirements**

- [x] Should be able to add date and time
- [x] Should be able to add start and end location
- [x] Should be able to select vehicle type
- [x] Cost allocation
  - [x] Should be able to select general cost assignment
  - [x] Should also be able to choose another cost allocation
- [x] Flex fields
  - [x] Should be able to add kilometers  
  - [x] Should be able to add amount of passengers
  - [x] Should be able to add weight of baggages
  - [x] Should be able to select country or region

- [x] Should be able to edit mileage
- [x] Should be able to delete mileage 
- [x] Information should be synced to SAP 
- [x] Information should be read from SAP if the user opens a travel from SAP

**Useful information for developer**
- Available vehicle types can be found with `/VehicleTypeSet` endpoint
- Available flex fields can be found with `/FlexibleMileageFieldSet` endpoint

**Important for tester**

- Flex fields might not be available for you. In my case, only kilometers and amount of passengers are available
- I had to make some changes to cost distribution to get it to work with mileage. Cost Distribution for travel should therefore also be tested to make sure it still works as expected

### PR DESCRIPTION
**Pull request by tormodAase**
This PR adds mileage to travel expense.

Closes #86

### CODE REVIEW

File: screens/NewTravelExpense/RefundsTab/components/MileageCard.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            <MaterialCommunityIcons
+              name="transit-transfer"
+              size={24}
+              color="#6F6F6F"
+            />
+          </View>
+          <View style={{ paddingLeft: 16 }}>
+            <Typography
+              medium
+              numberOfLines={1}
```
Comment (dbjoEq) [inline screens/NewTravelExpense/RefundsTab/components/MileageCard.tsx:32]: I get this error on numberOfLines:

`Property 'numberOfLines' does not exist on type`

File: types.tsx [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 import { MaterialCommunityIcons, MaterialIcons } from '@expo/vector-icons';
 import { DateTime } from 'luxon';
-import { TravelPeriod } from './store/storeModel';
+import { CostObject, Identifier, TravelPeriod } from './store/storeModel';
```
Comment (dbjoEq) [inline types.tsx:8]: This seems to cause merge conflict.

File: screens/NewTravelExpense/RefundsTab/components/MileageCard.tsx (hunk 1 above)
Comment (tormodAase) [reply screens/NewTravelExpense/RefundsTab/components/MileageCard.tsx:32]: Should be fixed now, please check

**Review by dbjoEq — DISMISSED** [review]
Looks good. 👍

**Comment by tormodAase** [comment]
There are still some things to test, but it might take a few days to get the access we need in order to test it. I'm therefore squashing and merging, and if anything else pops up, Natalie will create a new issue

**Pull request merged**